operator.md

```diff
@@ -16,18 +16,26 @@ kubectl create clusterrolebinding kubeflow-operator --clusterrole cluster-admin
 kubectl create -f deploy/operator.yaml -n ${OPERATOR_NAMESPACE}
 ```
 
-2. Deploy KfDef. You can optionally apply ResourceQuota if your Kubernetes version is 1.15+, which will allow only one _kfdef_ instance or one deployment of Kubeflow on this cluster, which follows the singleton model.
+2. Setup Kubeflow namespace. You can optionally apply ResourceQuota if your Kubernetes version is 1.15+, which will allow only one _kfdef_ instance or one deployment of Kubeflow on this cluster, which follows the singleton model.
 we use ResourceQuota to provide constraints that only one instance of kfdef is allowed within the Kubeflow namespace.
 ```shell
 KUBEFLOW_NAMESPACE=kubeflow
 kubectl create ns ${KUBEFLOW_NAMESPACE}
 # kubectl create -f deploy/crds/kfdef_quota.yaml -n ${KUBEFLOW_NAMESPACE} # only deploy this if the k8s cluster is 1.15+ and has resource quota support
-kubectl create -f <kfdef> -n ${KUBEFLOW_NAMESPACE}
 ```
 
-_<kfdef>_ above can point to a remote URL or to a local kfdef file. For e.g. for IBM Cloud, command will be
+3. Deploy KfDef.
+```
+kubectl create -f <kfdef> -n ${KUBEFLOW_NAMESPACE}
+```
+_kfdef_ above can point to a remote URL or to a local kfdef file. If you want to use the set of default kfdefs from Kubeflow. You will have to insert the `metadata.name` field before you can apply it to Kubernetes. Below are the commands for applying the KubeFlow 1.0 _kfdef_ using Operator. For e.g. for IBM Cloud, commands will be
 ```shell
-kubectl create -f https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_ibm.yaml -n ${KUBEFLOW_NAMESPACE}
+export KUBEFLOW_DEPLOY_NAME=kubeflow
+export KFDEF_URL=https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_ibm.yaml
+export KFDEF=$(echo "${KFDEF_URL}" | rev | cut -d/ -f1 | rev)
+curl -L ${KFDEF_URL} > ${KFDEF}
+perl -pi -e $'s@metadata:@metadata:\\\n  name: '"${KUBEFLOW_DEPLOY_NAME}"'@' ${KFDEF}
```

Comment:
Do we need perl?

Reply:
If we are going to use sed, then the commands will be different between osx and linux. If we want to have the same commands for both platform, usually people do it with awk or perl.

Reply:
ok - and will this work for the case where the kfdef is not through a URL, but locally on file system?

Reply:
If you really want to use "perl" or "sed" or "awk" I won't stop you; but I really don't like the idea of using an unstructured approach to process structured data.

I much prefer using a tool like yq
https://github.com/mikefarah/yq

Even though it means people have to install a separate binary I think it is a much better solution for modifying yaml files.

Reply:
Thanks Jeremy, I updated the instructions to use yq. I kept the perl command in case someone is having trouble to install the yq binary.